### PR TITLE
1955 hermit proover integration

### PR DESCRIPTION
**Pull request by erithacus-rubecula**
requested pull for hermit integration branch.

### CODE REVIEW

**Comment by tillmo** [comment]
@mcodescu can you please review this one?

**Comment by mcodescu** [comment]
Yes.

**Comment by mcodescu** [comment]
The last problem reported by me in the ticket #1955 is not fully solved: I see that you have added the case to `analyseOutput` but the other change was not made and therefore I get the same error when trying to disprove something. I will mark the change I had to make directly in the code. 

You should also remove the tracing messages.

File: OWL2/ProveHermit.hs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+           : (if doEntail then ["--premise=file://"++timeTmpFile, "--conclusion=file://"++entFile, "-E"] 
+              else ["-k", timeTmpFile]) ++ words opts
+      case entail of
+        Just c -> writeFile entFile c
+        Nothing -> return ()
+      mex <- timeoutCommand secs "java" args
+      trace (show mex ++ "\n" ++ show args) (removeFile timeTmpFile)
+      removeFile timeTmpEnt
+      when doEntail $ removeFile entFile
+      return $ fmap (\ (_, outS, errS) -> if outS == "true\n" then (True, outS, errS) else (False, outS, errS)) mex
```
Comment (mcodescu) [inline OWL2/ProveHermit.hs:199]: Using False here make the tool report it cannot find the prover. It worked for me as expected with:
`return $ fmap (\ (_, outS, errS) -> (True, outS, errS)) mex`

**Review by mcodescu — CHANGES_REQUESTED** [review]

**Review by mcodescu — APPROVED** [review]
:+1:

**Pull request merged**